### PR TITLE
Doc: add documentation for headers

### PR DESCRIPTION
**Pull request by tusharmath**

### CODE REVIEW

**Comment by amitksingh1490** [comment]
## 🚀🚀🚀 Benchmark Results 
 **Concurrency: 256** 
 **Requests/sec: 432902.32**

File: zio-http/src/main/scala/zhttp/http/headers/HeaderChecks.scala [hunk 1]
```diff
@@ -3,7 +3,13 @@ package zhttp.http.headers
 import io.netty.util.AsciiString.contentEqualsIgnoreCase
 import zhttp.http.Headers.Literals.Value
 
-trait HeaderChecks { self: HeaderGetters =>
+/**
+ * Maintains a list of operators that checks if the Headers meet the give constraints.
```
Comment (ashprakasan) [inline zio-http/src/main/scala/zhttp/http/headers/HeaderChecks.scala]:
```suggestion
 * Maintains a list of operators that checks if the Headers satisfy the given constraints.
```

File: zio-http/src/main/scala/zhttp/http/headers/HeaderNames.scala [hunk 2]
```diff
@@ -2,6 +2,10 @@ package zhttp.http.headers
 
 import io.netty.handler.codec.http.HttpHeaderNames
 
+/**
+ * List of commonly use HeaderNames. They are provided to reduce bugs caused by typos and also to improve performance.
```
Comment (ashprakasan) [inline zio-http/src/main/scala/zhttp/http/headers/HeaderNames.scala]:
```suggestion
 * List of commonly used HeaderNames. They are provided to reduce bugs caused by typos and also to improve performance.
```

File: zio-http/src/main/scala/zhttp/http/headers/HeaderNames.scala [hunk 3]
```diff
@@ -2,6 +2,10 @@ package zhttp.http.headers
...
+ * `HeaderNames` arent encoded everytime one needs to send them over the wire.
```
Comment (ashprakasan) [inline zio-http/src/main/scala/zhttp/http/headers/HeaderNames.scala]:
```suggestion
 * `HeaderNames` aren't encoded every time it needs to be sent over the wire.
```

File: zio-http/src/main/scala/zhttp/http/headers/HeaderValues.scala [hunk 4]
```diff
@@ -2,6 +2,10 @@ package zhttp.http.headers
 
 import io.netty.handler.codec.http.HttpHeaderValues
 
+/**
+ * List of commonly use HeaderValues. They are provided to reduce bugs caused by typos and also to improve performance.
```
Comment (ashprakasan) [inline zio-http/src/main/scala/zhttp/http/headers/HeaderValues.scala]:
```suggestion
 * List of commonly used HeaderValues. They are provided to reduce bugs caused by typos and also to improve performance.
```

File: zio-http/src/main/scala/zhttp/http/headers/HeaderValues.scala [hunk 5]
```diff
@@ -2,6 +2,10 @@ package zhttp.http.headers
...
+ * `HeaderValues` arent encoded everytime one needs to send them over the wire.
```
Comment (ashprakasan) [inline zio-http/src/main/scala/zhttp/http/headers/HeaderValues.scala]:
```suggestion
 * `HeaderValues` arent encoded every time one needs to send them over the wire.
```

File: zio-http/src/main/scala/zhttp/http/headers/HeaderChecks.scala [hunk 6]
```diff
@@ -3,7 +3,13 @@ package zhttp.http.headers
...
+ *
+ * NOTE: Add methods here, if it tests the Headers for something, and returns a true or false based on if the conditions
```
Comment (ashprakasan) [inline zio-http/src/main/scala/zhttp/http/headers/HeaderChecks.scala]:
```suggestion
 * NOTE: Add methods here, if it tests the `Headers` for something, and returns `true` or `false` depending on whether or not the conditions are met.
```

File: zio-http/src/main/scala/zhttp/http/headers/HeaderChecks.scala [hunk 7]
```diff
@@ -3,7 +3,13 @@ package zhttp.http.headers
...
+ * are met or not.
```
Comment (ashprakasan) [inline zio-http/src/main/scala/zhttp/http/headers/HeaderChecks.scala]:
```suggestion

```

File: zio-http/src/main/scala/zhttp/http/headers/HeaderModifier.scala [hunk 8]
```diff
@@ -3,7 +3,14 @@ package zhttp.http.headers
 import zhttp.http.{Cookie, Header, Headers, Method}
 import zio.duration.Duration
 
-trait HeaderModifier[+A] { self =>
+/**
+ * Maintains a list of operators that modify the current Headers. Once modified, a new instance of the same type is
+ * returned. So or eg: `request.addHeader("A", "B")` should return a new `Request` and similarly `headers.add("A", "B")`
```
Comment (ashprakasan) [inline zio-http/src/main/scala/zhttp/http/headers/HeaderModifier.scala]:
```suggestion
 * returned. For eg: `request.addHeader("A", "B")` should return a new `Request` and similarly `headers.add("A", "B")`
```

**Comment by amitksingh1490** [comment]
## 🚀🚀🚀 Benchmark Results 
 **Concurrency: 256** 
 **Requests/sec: 436150.61**

**Comment by tusharmath** [comment]
closing in favour of #684

**Pull request closed without merging**